hy/contrib/slicing.hy

```diff
@@ -0,0 +1,193 @@
+;;; Hy Multi-Index Slicing module
+;; Copyright 2021 the authors.
+;; this file is part of Hy, which is free software licensed under the Expat
+;; license. See the LICENSE.
+"Macros for elegantly slicing and dicing multi-axis and multidimensional sequences.
+
+Libraries like ``numpy`` and ``pandas`` make extensive use of python's slicing syntax
+and even extends it to allow multiple slices using tuples. This makes handling
+multi-axis (pandas) and multidimensional arrays (numpy) extremely elegant and efficient.
+Hy doesn't support Python's sugared slicing syntax (``1::-2 # => slice(1, None, 2)``).
+Which makes slicing quite cumbersome, especially when tuple's get thrown into the mix.
+Hy's ``cut`` form makes single slices easy, but anything more than that becomes
+much more difficult to construct and parse. Where python can express a multidimensional
+slice as:
+
+.. code-block:: python
+
+   >>> arr[1:-2:2,3:5]
+
+The equivalent in Hy would be::
+
+   => (get arr (, (slice 1 -2 2) (slice 3 5)))
+
+which is hardly ideal. This module provides an ``ncut`` macro and a ``#|`` tag macro
+that enable python's sugared slicing form in Hy so that the previous could be expressed as::
+
+   => (ncut a 1:-1:2 3:5)
+
+or more manually using the tag macro as::
+
+   => (get a (, #| 1:-2:2 #| 3:5))
+"
+(eval-and-compile
+  (defn parse-colon [sym]
+    (lfor index (.split (str sym) ":")
```

Comment:
(Not sure how much we care about this but) since `HySymbol` already inherits from `str`, the call to `str` here is unnecessary

Reply:
the slice `:1:2` actually gets parsed as a `HyKeyword` which doesn't have `.split` so we can actually get a `HySymbol` or a `HyKeyword` here

Reply:
Ah right, because of the initial colon. Good point, my bad!

Reply:
Also, the fact that `HySymbol` inherits from `str`, instead of containing a `str` as an attribute like `HyKeyword`, is an implementation detail that I wouldn't rely on.